pkg/asset/manifests/tectonic.go

```diff
@@ -78,6 +116,18 @@ func (t *Tectonic) Generate(dependencies asset.Parents) error {
 		"99_tectonic-system-02-pull.json":                        applyTemplateData(content.PullTectonicSystem, templateData),
 	}
 
+	switch platform {
+	case "aws", "openstack":
+		conditionalAssetData := map[string][]byte{
+			"99_cloud-creds-secret.yaml":             applyTemplateData(content.CloudCredsSecret, templateData),
+			"99_role-cloud-creds-secret-reader.yaml": applyTemplateData(content.RoleCloudCredsSecretReader, templateData),
+		}
+
+		for file, content := range conditionalAssetData {
+			assetData[file] = content
```

Comment:
`conditionalAssetData` seems like more trouble than it's worth.  Why not just:

```go
assetData["99_..."] = applyTemplate...
```

Reply:
because doh-duh thank you!